### ISSUE TITLE
Introduce Image field

### ISSUE THREAD

**Issue by eduardoboucas** [dadi/api#415]
The current method for referencing an image is as follows:

```json
"leadImage": {
  "type": "Reference",
  "settings": {
    "collection": "mediaStore"
  }
}
```

I think this is confusing, because `mediaStore` is actually a magic string that maps to an internal collection and therefore users don't need a schema for it. With the modularisation of field types introduced in 3.1, I think the above should be changed to:

```json
"leadImage": {
  "type": "Image"
}
```

The `settings` block could describe the meta fields that are to be stored with the image.

Thoughts @jimlambie?

**Comment by jimlambie** [comment]
@eduardoboucas 😍 

I agree - let's bring these features out into the open.

**Comment by jimlambie** [comment]
We could make it less breaking, by keeping both forms and deprecating the old one at some point,

**Comment by mingard** [comment]
Still keeping the settings block collection flag right?

**Comment by eduardoboucas** [comment]
We could keep it for a period of time like @jimlambie suggested, but the idea is to remove it at some point. The `settings.collection` field will still be used for normal Reference fields, of course, but in the case of media we're asking people to reference a collection they haven't created.

**Comment by eduardoboucas** [comment]
You're absolutely right. That somehow slipped out of my mind!

The `Image` type could allow people to omit the `settings` block _if_ they wanted to use the default media bucket, which like you said will often be the case. We could then allow them to change the media bucket by supplying an optional `settings.mediaBucket` property (supporting `settings.collection` for backward-compatibility).

I know these changes may seem irrelevant, but I really think that changing the semantics around media would help make it a first-class citizen in API.

**Comment by mingard** [comment]
LGTM

**Comment by danwdart** [comment]
This sounds like a really awesome suggestion - I'll change stuff to this when needed.

**Comment by eduardoboucas** [comment]
Closed in #504.
